llvm/tools/spir-to-ir-wrapper/spir-to-ir-wrapper.cpp

```diff
@@ -0,0 +1,142 @@
+//
+// This utility checks if the input file is SPIR based.  If so, convert to IR
+// The input can be either SPIR-V or LLVM-IR.  When LLVM-IR, copy the file to
+// the specified output.
+//
+// Uses llvm-spirv to perform the conversion if needed.
+//
+// The output file is used to allow for proper input and output flow within
+// the driver toolchain.
+//
+// Usage: spir-to-ir-wrapper input.spv -o output.bc
+//
+//===----------------------------------------------------------------------===//
+
+#include "LLVMSPIRVLib.h"
+#include "llvm/BinaryFormat/Magic.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/InitLLVM.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Program.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/StringSaver.h"
+
+using namespace llvm;
+
+// InputFilename - The filename to read from.
+static cl::opt<std::string> InputFilename(cl::Positional,
+                                          cl::value_desc("<input spv file>"),
+                                          cl::desc("<input file>"));
+
+// Output - The filename to output to.
+static cl::opt<std::string> Output("o", cl::value_desc("output IR filename"),
+                                   cl::desc("output filename"));
+
+// LlvmSpirvOpts - The filename to output to.
+static cl::opt<std::string>
+    LlvmSpirvOpts("llvm-spirv-opts", cl::value_desc("llvm-spirv options"),
+                  cl::desc("options to pass to llvm-spirv"));
+
+static void error(const Twine &Message) {
+  llvm::errs() << "spir-to-ir-wrapper: " << Message << '\n';
+  exit(1);
+}
+
+// Convert the SPIR-V to LLVM-IR.
+static int convertSPIRVToLLVMIR(const char *Argv0) {
+  // Find llvm-spirv.  It is expected this resides in the same directory
+  // as spir-to-ir-wrapper.
+  StringRef ParentPath = llvm::sys::path::parent_path(Argv0);
+  llvm::ErrorOr<std::string> LlvmSpirvBinary =
+      llvm::sys::findProgramByName("llvm-spirv", ParentPath);
+  if (!LlvmSpirvBinary)
+    LlvmSpirvBinary = llvm::sys::findProgramByName("llvm-spirv");
+
+  SmallVector<StringRef, 6> LlvmSpirvArgs = {"llvm-spirv", "-r", InputFilename,
+                                             "-o", Output};
+
+  // Add any additional options specified by the user.
+  SmallVector<const char *, 8> TargetArgs;
+  llvm::BumpPtrAllocator BPA;
+  llvm::StringSaver S(BPA);
+  if (!LlvmSpirvOpts.empty()) {
+    // Tokenize the string.
+    llvm::cl::TokenizeGNUCommandLine(LlvmSpirvOpts, S, TargetArgs);
+    std::copy(TargetArgs.begin(), TargetArgs.end(),
+              std::back_inserter(LlvmSpirvArgs));
+  }
+
+  return llvm::sys::ExecuteAndWait(LlvmSpirvBinary.get(), LlvmSpirvArgs);
+}
+
+static int copyInputToOutput(void) {
+  // When given an output file, just copy the input to the output
+  if (!Output.empty() && !InputFilename.empty()) {
+    llvm::sys::fs::copy_file(InputFilename, Output);
+  }
+  return 0;
+}
```

Comment:
Since the presence of legal input/output file names has already been checked in main and there's no subsequent else block for a duplicating assertion, I'd suggest simply removing the condition. Altogether, it would come down to:
```suggestion
static int copyInputToOutput(void) {
  return llvm::sys::fs::copy_file(InputFilename, Output);
}
```